core/src/main/java/org/bitcoinj/coinjoin/utils/CoinJoinTransactionType.java

```diff
@@ -47,24 +49,43 @@ public static CoinJoinTransactionType fromTx(Transaction tx, TransactionBag tran
                         // <case2>, see CCoinJoinClientSession.makeCollateralAmounts
                         (nAmount0 == nAmount1 && CoinJoin.isCollateralAmount(nAmount0));
             } else if (tx.getOutputs().size() == 1) {
-                // <case3>, see CCoinJoinClientSession.makeCollateralAmounts
-                makeCollateral = CoinJoin.isCollateralAmount(tx.getOutput(0).getValue());
+                TransactionOutput firstOutput = tx.getOutput(0);
+
+                if (CoinJoin.isCollateralAmount(firstOutput.getValue())) {
+                    // <case3>, see CCoinJoinClientSession.makeCollateralAmounts
+                    makeCollateral = true;
+                } else if (tx.getInputs().size() > 1 && tx.getValue(transactionBag).plus(tx.getFee()).equals(Coin.ZERO)) {
```

Comment:
if `tx.getFee()` is `null`, which will be true for received transactions that have one output, then an NPE exception will be thrown in `Coin.plus`.